src/include/simeng/RegisterValue.hh

```diff
@@ -26,10 +26,16 @@ class RegisterValue {
    * number of bytes (defaulting to the size of the template type). */
   template <class T,
             typename std::enable_if_t<!std::is_pointer_v<T>, T>* = nullptr>
-  RegisterValue(T value, uint16_t bytes = sizeof(T)) : bytes(bytes) {
+  RegisterValue(T value, uint16_t bytes = sizeof(T), bool relaxFor32 = true) : bytes(bytes) {
```

Comment:
I think the below statement should be added to this constructor to avoid the case when the bytes argument provided is less than `sizeof(T)`

```
bytes = std::max(bytes, sizeof(T));
```
I think having bytes being less than `sizeof(T)` doesn't make sense and is dangerous; As it will lead to data truncation. Which I'm not sure why would be needed if a smaller data type can be used.  Even if something like this is intentional I still think it should not be allowed.

Reply:
A similar statement should be added in the constructor `  RegisterValue(const char* ptr, uint16_t bytes, uint16_t capacity)`

Reply:
Please[ see](https://github.com/UoB-HPC/SimEng/pull/317/files#r1309948928)